presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -154,9 +156,19 @@ protected Node visitShowStats(ShowStats node, Void context)
                 throw rewriteAccessDeniedException(e);
             }
 
+            for (ColumnHandle column : columnHandles.values()) {
+                ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, column);
```

Comment:
i think you should grab `ConnectorTableMetadata` instead of calling connector metadata for each column separately.